.github/workflows/rolling-release.yml

```diff
@@ -3,332 +3,282 @@ name: Create rolling release
+    steps:
+      - uses: actions/checkout@v2
         with:
-          name: "linux-x64-appimage-qt"
+          fetch-depth: 0
+
+      - name: Compile with Gradle
+        working-directory: android
+        run: ./gradlew assembleRelease
 
-      - name: Download Qt AppImage zsync Artifact
-        uses: actions/download-artifact@v1
+      - name: Sign APK
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev'
+        uses: r0adkll/sign-android-release@v1
+        with:
+          releaseDirectory: android/app/build/outputs/apk/release
+          signingKeyBase64: ${{ secrets.APK_SIGNING_KEY }}
+          alias: ${{ secrets.APK_KEY_ALIAS }}
+          keyStorePassword: ${{ secrets.APK_KEY_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.APK_KEY_PASSWORD }}
+
+      - name: Rename APK
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev'
+        run: mv android/app/build/outputs/apk/release/app-release-unsigned-signed.apk duckstation-android.apk
+          
+      - name: Upload APK
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev'
+        uses: actions/upload-artifact@v2
         with:
-          name: "linux-x64-appimage-qt-zsync"
+          name: android
+          path: duckstation-android.apk
+          if-no-files-found: error
 
-      - name: Download Android APK
-        uses: actions/download-artifact@v1
+  macos-build:
+    runs-on: macos-10.15
+    steps:
+      - uses: actions/checkout@v2
         with:
-          name: "android"
+          fetch-depth: 0
+
+      - name: Install packages
+        env:
+          HOMEBREW_NO_ANALYTICS: 1
+        run: brew install qt5 sdl2
 
-      - name: Download Mac App
-        uses: actions/download-artifact@v1
+      - name: Clone mac externals
+        uses: actions/checkout@v2
         with:
-          name: "macos-x64"
+          repository: stenzek/duckstation-ext-mac
+          path: dep/mac
+
+      - name: Compile and archive .app
+        env:
+          MACOSX_DEPLOYMENT_TARGET: 10.14
+        run: |
+          mkdir build
+          cd build
+          cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SDL_FRONTEND=OFF -DBUILD_QT_FRONTEND=ON -DUSE_SDL2=ON -DQt5_DIR=/usr/local/opt/qt/lib/cmake/Qt5 ..
+          cmake --build . -j $(sysctl -n hw.activecpu)
+          tar -C bin -czf ../duckstation-mac-release.tar.gz DuckStation.app
+
+      - name: Upload macOS archive
+        uses: actions/upload-artifact@v2
+        with:
+          name: macos-x64
+          path: duckstation-mac-release.tar.gz
+          if-no-files-found: error
+
+
+  create-release:
+    needs: [windows-build, windows-arm64-build, linux-build, android-build, macos-build]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/dev'
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v2
+
+      - name: Create Windows release archives
```

Comment:
I'm not sure if uploading all files and then downloading/recompressing is the best idea here. I guess it makes downloading the artifacts easier if you want to back to old builds...

Reply:
If you upload a zip as an artifact it gets double-zipped.

Reply:
Yeah, figured as much. My concern was mainly the final zip happening on a linux runner vs a windows runner originally (permissions etc).